plugins/opusCodec/src/OpusDecoder.cpp

```diff
@@ -0,0 +1,121 @@
+#include <PerfStat.h>
+#include <QtCore/QLoggingCategory>
+#include <AudioConstants.h>
+
+#include "OpusDecoder.h"
+
+static QLoggingCategory decoder("AthenaOpusDecoder");
+
+static QString error_to_string(int error) {
+    switch (error) {
+    case OPUS_OK:
+        return "OK";
+    case OPUS_BAD_ARG:
+        return "One or more invalid/out of range arguments.";
+    case OPUS_BUFFER_TOO_SMALL:
+        return "The mode struct passed is invalid.";
+    case OPUS_INTERNAL_ERROR:
+        return "An internal error was detected.";
+    case OPUS_INVALID_PACKET:
+        return "The compressed data passed is corrupted.";
+    case OPUS_UNIMPLEMENTED:
+        return "Invalid/unsupported request number.";
+    case OPUS_INVALID_STATE:
+        return "An encoder or decoder structure is invalid or already freed.";
+    default:
+        return QString("Unknown error code: %i").arg(error);
+    }
+}
+
+
+AthenaOpusDecoder::AthenaOpusDecoder(int sampleRate, int numChannels)
+{
+    int error;
+
+    _opus_sample_rate = sampleRate;
+    _opus_num_channels = numChannels;
+
+    _decoder = opus_decoder_create(sampleRate, numChannels, &error);
+
+    if ( error != OPUS_OK ) {
+        qCCritical(decoder) << "Failed to initialize Opus encoder: " << error_to_string(error);
+        _decoder = nullptr;
+        return;
+    }
+
+
+    qCDebug(decoder) << "Opus decoder initialized, sampleRate = " << sampleRate << "; numChannels = " << numChannels;
+}
+
+AthenaOpusDecoder::~AthenaOpusDecoder()
+{
+    if ( _decoder )
+        opus_decoder_destroy(_decoder);
+
+}
+
+void AthenaOpusDecoder::decode(const QByteArray &encodedBuffer, QByteArray &decodedBuffer)
+{
```

Comment:
Please put { on same line as function name CS 4.1.4 https://github.com/kasenvr/project-athena/blob/kasen/core/CODING_STANDARD.md#414-method-definitions-should-have-the-following-form